pkg/util/testing/client.go

```diff
@@ -18,22 +18,42 @@ package testing
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"strings"
 	"sync"
 
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
 )
 
```

Comment:
I wold just go with something like: 

```go
type fakeStrategicMegePatc struct {
	p client.Patch
}

func (*fakeStrategicMegePatc) Type() types.PatchType {
	return types.StrategicMergePatchType
}
func (fp *fakeStrategicMegePatc) Data(obj client.Object) ([]byte, error) {
	return fp.p.Data(obj)
}

func SSAToSM(ctx context.Context, clnt client.Client, SubResourceName string, obj client.Object, patch client.Patch, opts ...client.SubResourcePatchOption) error {
	if patch.Type() == types.ApplyPatchType {
		patch = &fakeStrategicMegePatc{p: patch}
	}
	return clnt.SubResource(SubResourceName).Patch(ctx, obj, patch, opts...)
}
```
Then in the tests that require this , use this function as 

```go
			manageBuilder = manageBuilder.WithInterceptorFuncs(interceptor.Funcs{SubResourcePatch: utiltesting.SSAToSM})
```
Note: It cannot be just added in 'NewClientBuilder' since:
 -  We don't need it in all the tests.
 -  `WithInterceptorFuncs`  that might be needed after `NewClientBuilder` return will replace it.

Reply:
We don't declare manageBuilder anywhere other than in multikueue.

Reply:
That is just an usage example